controllers/clusters/cadence_controller_test.go

```diff
@@ -0,0 +1,259 @@
+			gomega.Expect(k8sClient.Create(ctx, cadenceManifest)).Should(gomega.Succeed())
+
+			key := client.ObjectKeyFromObject(cadenceManifest)
+			cadence := &v1beta1.Cadence{}
+
+			expectedClusterID := cadenceManifest.Spec.Name + "-" + mock.CreatedID
+
+			gomega.Eventually(func() (string, error) {
+				if err := k8sClient.Get(ctx, key, cadence); err != nil {
+					return "", err
+				}
+
+				return cadence.Status.ID, nil
+			}, timeout, interval).Should(gomega.Equal(expectedClusterID))
+		})
+	})
+
+	ginkgo.When("deleting cadence resource", func() {
+		ginkgo.It("should successfully delete cadence resource in k8s and Instaclustr", func() {
+			cadenceManifest := *cadenceManifest
+			cadenceManifest.ResourceVersion = ""
+			cadenceManifest.Name += "-deleting"
+			cadenceManifest.Spec.Name += "-deleting"
+
+			gomega.Expect(k8sClient.Create(ctx, &cadenceManifest)).Should(gomega.Succeed())
+
+			key := client.ObjectKeyFromObject(&cadenceManifest)
+			cadence := &v1beta1.Cadence{}
+
+			gomega.Eventually(func(g gomega.Gomega) {
+				g.Expect(k8sClient.Get(ctx, key, cadence)).Should(gomega.Succeed(), "should get the cadence resource")
+				g.Expect(cadence.Status.ID).ShouldNot(gomega.BeEmpty(), "the cadence id should not be empty")
+			}, timeout, interval).Should(gomega.Succeed())
+
+			gomega.Expect(k8sClient.Delete(ctx, cadence)).Should(gomega.Succeed())
+
+			gomega.Eventually(func(g gomega.Gomega) {
+				c := &v1beta1.Cadence{}
+				g.Expect(k8sClient.Get(ctx, key, c)).ShouldNot(gomega.Succeed())
+			}, timeout, interval).Should(gomega.Succeed())
+
+			gomega.Eventually(func() error {
+				_, err := instaClient.GetCadence(cadence.Status.ID)
+				return err
+			}, timeout, interval).Should(gomega.Equal(instaclustr.NotFound))
+		})
+	})
+
+	ginkgo.When("apply valid cadence manifest with packaged provisioning", func() {
+		ginkgo.It("should create cadence and all packaged resources in k8s and on Instaclustr", func() {
+			cadenceManifest := *cadenceManifest
+			cadenceManifest.ResourceVersion = ""
+			cadenceManifest.Spec.StandardProvisioning = nil
+			cadenceManifest.Spec.Name += "-packaged-provisioning"
+			cadenceManifest.Name += "-packaged-provisioning"
+			cadenceManifest.Spec.PackagedProvisioning = []*v1beta1.PackagedProvisioning{
+				{
+					UseAdvancedVisibility: true,
+					BundledKafkaSpec: &v1beta1.BundledKafkaSpec{
+						NodeSize:          "test-kafka-node-size-1",
+						NodesNumber:       2,
+						Network:           "10.0.0.0/16",
+						ReplicationFactor: 2,
+						PartitionsNumber:  2,
+					},
+					BundledOpenSearchSpec: &v1beta1.BundledOpenSearchSpec{
+						NodeSize:          "test-opensearch-node-size-1",
+						ReplicationFactor: 2,
+						Network:           "10.1.0.0/16",
+					},
+					BundledCassandraSpec: &v1beta1.BundledCassandraSpec{
+						NodeSize:          "test-cassandra-node-size-1",
+						NodesNumber:       2,
+						ReplicationFactor: 2,
+						Network:           "10.2.0.0/16",
+					},
+				},
+			}
```

Comment:
I suggest you create a new data_test file for the packaged provisioning resources

Reply:
Added new data file for packaged provisioning